### PR TITLE
feat: added `assert/is-negative-finite`

### LINKED ISSUE #1345: [RFC]: add `@stdlib/assert/is-negative-finite`
### Description

This RFC proposes adding the package `@stdlib/assert/is-negative-finite`.

The package should be similar in structure to [`@stdlib/assert/is-negative-number`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-negative-number) and should have a similar implementation except negative infinity should evaluate to `false`.

```javascript
function isNegativeFinite( value ) { ... }
```

Package: `@stdlib/assert/is-negative-finite`
Alias: `isNegativeFinite`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
This commit adds the assert/is-negative-finite module for handling negative finite values.

Resolves #1345  .

## Description

> What is the purpose of this pull request?

Adds the package `@stdlib/assert/is-negative-finite`.

## Related Issues

> Does this pull request have any related issues?

Yes, it resolves #1345 

This pull request:

-   resolves #1345 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Test Logs:
Screenshot 2024-02-23 at 21 45 54

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+* @example
+* var bool = isNegativeFinite( new Number( Number.NEGATIVE_INFINITY ) );
+* // returns false
+*/
+function isNegativeFinite( value ) {
+	return (
+		isNumber( value ) &&
+		value < 0.0 && isFinite(value)
+	);
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js:56]: I think we need to import `is-finite` just like `var isNumber`

File: lib/node_modules/@stdlib/assert/is-negative-finite/package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  "scripts": {},
+  "homepage": "https://github.com/stdlib-js/stdlib",
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/stdlib-js/stdlib.git"
+  },
+  "bugs": {
+    "url": "https://github.com/stdlib-js/stdlib/issues"
+  },
+  "dependencies": {},
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-negative-finite/package.json:34]: It depends on `is-finite`

Comment (Planeshifter) [reply lib/node_modules/@stdlib/assert/is-negative-finite/package.json:34]: No reason to manually populate the `dependencies` field; that is all automated in the publishing pipeline.

File: lib/node_modules/@stdlib/assert/is-negative-finite/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/README.md]:
```suggestion
Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/benchmark/benchmark.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2019 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/index.d.ts]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2019 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	isNegativeFinite.isPrimitive( -2 ); // $ExpectType boolean
+}
+
+// The compiler throws an error if the isPrimitive method is provided an unsupported number of arguments...
+{
+	isNegativeFinite.isPrimitive(); // $ExpectError
+	isNegativeFinite.isPrimitive( -2, 123 ); // $ExpectError
+}
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/examples/index.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/examples/index.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/index.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/lib/index.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/main.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/lib/main.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/test/test.main.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/test/test.main.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/test/test.js [hunk 12]
```diff
@@ -0,0 +1,43 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/test/test.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js [hunk 13]
```diff
@@ -0,0 +1,60 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/test/test.object.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/test/test.object.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/test/test.primitive.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/test/test.primitive.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

**Review by Planeshifter — COMMENTED** [review]
This should be close to being able to get merged, thanks for your contribution!

As @Pranavchiku mentioned, we should explicitly require [@stdlib/assert/is-finite](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-finite).

File: lib/node_modules/@stdlib/assert/is-negative-finite/README.md [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+
+## Examples
+
+<!-- eslint-disable no-new-wrappers -->
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var Number = require( '@stdlib/number/ctor' );
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/README.md:108]: Need to require the package again inside the examples code block of the `README.md` file:

```suggestion
var isNegativeFinite = require( '@stdlib/assert/is-negative-finite' );

```

File: lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+// Attached to main export is an isPrimitive method which returns a boolean...
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/docs/types/test.ts]:
```suggestion
// Attached to main export is an isObject method which returns a boolean...
```

File: lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js:56]: cc @gunjjoshi 
```diff
--- a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js
+++ b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js
@@ -21,6 +21,7 @@
 // MODULES //
 
 var isNumber = require( '@stdlib/assert/is-number' ).isObject;
+var isFinite = require( '@stdlib/assert/is-finite' ).isObject;
 
 
 // MAIN //
diff --git a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
index 02f04272b8..6e40cf290d 100644
--- a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
+++ b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
@@ -21,6 +21,7 @@
 // MODULES //
 
 var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isFinite = require( '@stdlib/assert/is-finite' ).isPrimitive;
 
 
 // MAIN //
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js:56]: I see here is the error:

```console
make[1]: Entering directory '/Users/pranavchiku/repos/stdlib-js/stdlib'

Linting file: lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js

/Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js
  24:5  error  isFinite is already defined  stdlib/no-redeclare

✖ 1 problem (1 error, 0 warnings)

make[1]: *** [/Users/pranavchiku/repos/stdlib-js/stdlib/tools/make/lib/lint/javascript/eslint.mk:225: eslint-files] Error 1
make[1]: Leaving directory '/Users/pranavchiku/repos/stdlib-js/stdlib'
make: *** [/Users/pranavchiku/repos/stdlib-js/stdlib/tools/make/lib/lint/javascript/Makefile:158: lint-javascript-files] Error 2

JavaScript lint errors for source files.

```

To fix this use the following diff

```diff
--- a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js
+++ b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/object.js
@@ -21,6 +21,7 @@
 // MODULES //
 
 var isNumber = require( '@stdlib/assert/is-number' ).isObject;
+var isFinite = require( '@stdlib/assert/is-finite' ).isObject; // eslint-disable-line stdlib/no-redeclare
 
 
 // MAIN //
diff --git a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
index 02f04272b8..e7356daee4 100644
--- a/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
+++ b/lib/node_modules/@stdlib/assert/is-negative-finite/lib/primitive.js
@@ -21,6 +21,7 @@
 // MODULES //
 
 var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isFinite = require( '@stdlib/assert/is-finite' ).isPrimitive; // eslint-disable-line stdlib/no-redeclare
 ```

**Comment by gunjjoshi** [comment]
Thanks for your suggestions @Planeshifter @Pranavchiku, I have made the suggested changes.

**Comment by Planeshifter** [comment]
@gunjjoshi CI has passed; will give it one more review and then merge. Thanks for your contribution!

File: lib/node_modules/@stdlib/assert/is-negative-finite/docs/repl.txt [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+        Boolean indicating whether value is a number object having a finite
+        negative value.
+
+    Examples
+    --------
+    > var bool = {{alias}}.isObject( -3.0 )
+    false
+    > bool = {{alias}}.isObject( new Number( -3.0 ) )
+    true
+    > var bool = {{alias}}.isObject( Number.NEGATIVE_INFINITY )
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-negative-finite/docs/repl.txt]:
```suggestion
    > bool = {{alias}}.isObject( Number.NEGATIVE_INFINITY )
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**